### PR TITLE
FOGL-2616 system tests added for ping, restart, shutdown added (admin api)

### PR DESCRIPTION
**Pull request by ashish-jabble**
**O/P**
```
$ pytest -s -vv test_common.py 
============================================================= test session starts =============================================================
platform linux -- Python 3.5.3, pytest-3.6.0, py-1.8.0, pluggy-0.6.0 -- /usr/local/bin/python3.5
cachedir: ../.pytest_cache
rootdir: /home/foglamp/Development/FogLAMP/tests/system/python, inifile: pytest.ini
plugins: mock-1.10.0, cov-2.5.1, asyncio-0.8.0, allure-adaptor-1.7.10, aiohttp-0.3.0
collected 5 items                                                                                                                             

../TestCommon test_ping_default <- api/test_common.py FogLAMP killed.
This script will remove all data stored in the SQLite3 datafile '/home/foglamp/Development/FogLAMP/data/foglamp.db'
Enter YES if you want to continue: Starting FogLAMP v1.5.1......
FogLAMP started.
FogLAMP v1.5.1 running.
FogLAMP Uptime:  6} seconds.
FogLAMP records: 0 read, 0 sent, 0 purged.
FogLAMP does not require authentication.
=== FogLAMP services:
foglamp.services.core
=== FogLAMP tasks:
PASSED
../TestCommon test_ping_when_auth_mandatory_allow_ping_true <- api/test_common.py PASSED
../TestCommon test_ping_when_auth_mandatory_allow_ping_false <- api/test_common.py FogLAMP killed.
This script will remove all data stored in the SQLite3 datafile '/home/foglamp/Development/FogLAMP/data/foglamp.db'
Enter YES if you want to continue: Starting FogLAMP v1.5.1......
FogLAMP started.
FogLAMP v1.5.1 running.
FogLAMP Uptime:  6,"dataRead" seconds.
FogLAMP records: 0 read, 0 sent, 0 purged.
FogLAMP does not require authentication.
=== FogLAMP services:
foglamp.services.core
=== FogLAMP tasks:

PASSED
../TestCommon test_restart <- api/test_common.py PASSED
../TestCommon test_shutdown <- api/test_common.py FogLAMP killed.
This script will remove all data stored in the SQLite3 datafile '/home/foglamp/Development/FogLAMP/data/foglamp.db'
Enter YES if you want to continue: Starting FogLAMP v1.5.1......
FogLAMP started.
FogLAMP v1.5.1 running.
FogLAMP Uptime:  6,"dataRead" seconds.
FogLAMP records: 0 read, 0 sent, 0 purged.
FogLAMP does not require authentication.
=== FogLAMP services:
foglamp.services.core
=== FogLAMP tasks:
PASSED

=== 5 passed in 85.71 seconds ==
```

### CODE REVIEW

File: tests/system/python/api/test_common.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        assert 'FogLAMP' == jdoc['serviceName']
+        assert host_name == jdoc['hostName']
+        assert ip_addresses == jdoc['ipAddresses']
+        assert 'green' == jdoc['health']
+        assert jdoc['authenticationOptional'] is True
+        assert jdoc['safeMode'] is False
+
+    @pytest.mark.skip(reason='Not Implemented yet')
+    def test_ping_when_auth_mandatory(self):
+        # TODO: Its a bit tricky to automate user input on shell when foglamp start
```
Comment (praveen-garg) [inline tests/system/python/api/test_common.py]: can not we set config auth mandatory and call  `conn.request("PUT", '/foglamp/restart')` and verify: allowPing True and False cases.

Comment (ashish-jabble) [reply tests/system/python/api/test_common.py]: Nope its 403 Forbidden
And syslog says `WARNING: common: foglamp.services.core.api.common: Permission denied for Ping when Auth is mandatory.`

File: tests/system/python/api/test_common.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import pytest
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.fixture
```
Comment (praveen-garg) [inline tests/system/python/api/test_common.py]: does this really need to be a fixture?

File: tests/system/python/api/test_common.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+        time.sleep(wait_time * 2)
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/ping')
+        r = conn.getresponse()
+        assert 403 == r.status
+        assert 'Forbidden' == r.reason
+
+    def test_restart(self):
+        assert 1, "Already verified in test_ping_when_auth_mandatory_allow_ping_true"
```
Comment (praveen-garg) [inline tests/system/python/api/test_common.py]: Already checked with test_ping_when_auth_mandatory_allow_ping_true, test_ping_when_auth_mandatory_allow_ping_false

**Pull request merged**